src/core/image.js

```diff
@@ -506,6 +506,122 @@ var PDFImage = (function PDFImageClosure() {
       }
     },
 
+    /**
+     * Check whether resize image data or not
+     * @param {Number} comps Number of color components, 1 or 3 is supported.
+     * @param {Number} bpc Number of bits per component.
+     * @return {Boolean}
+     */
+    shallResizeImage: function PDFImage_shallResizeImage(comps, bpc) {
+        return (comps === 1 && (bpc === 1 || bpc === 8));
+    },
+
+    /**
+     * Resize black white image data
+     * @param {Uint8Array} imgData image data.
+     * @param {Number} scaleBits
+     * @author Ramsoft
+     */
+    resizeBWPixels: function PDFImage_resizeBWPixels(imgData, scaleBits) {
+      var newWidth = imgData.width >> scaleBits;
+      var newHeight = imgData.height >> scaleBits;
+
+      var newRowBytes = (newWidth + 7) >> 3;
+      var originalRowBytes = (imgData.width + 7) >> 3;
+      var step = 1 << scaleBits;
+      var numBytes = newHeight * newRowBytes;
+      var pixelArrayOutput = new Uint8Array(numBytes);
+      var originalRowStart;
+      var newRowStart;
+
+      for (var i = 0, y2 = 0; i < newHeight; i++, y2 += step,
```

Comment:
This `for` loop condition is hard to understand. Please split this into multiple variables/lines.